vavr/src/main/java/io/vavr/collection/TreeSet.java

```diff
@@ -532,17 +532,16 @@
     @Override
     public TreeSet<T> addAll(Iterable<? extends T> elements) {
         Objects.requireNonNull(elements, "elements is null");
+        if (Collections.isEmpty(elements) || this.equals(elements)){
```

Comment:
A TreeSet is a SortedSet. We can't check for equality here because insertion depends on the underlying Comparator. Please restore the previous version.